cns/api.go

```diff
@@ -44,6 +48,42 @@ type DeleteNetworkRequest struct {
 	NetworkName string
 }
 
+// CreateHnsNetworkRequest describes request to create the HNS network.
+type CreateHnsNetworkRequest struct {
+	NetworkName          string            `json:",omitempty"`
+	NetworkType          string            `json:",omitempty"`
+	NetworkAdapterName   string            `json:",omitempty"`
+	SourceMac            string            `json:",omitempty"`
+	Policies             []json.RawMessage `json:",omitempty"`
+	MacPools             []MacPool         `json:",omitempty"`
+	Subnets              []SubnetInfo      `json:",omitempty"`
+	DNSSuffix            string            `json:",omitempty"`
+	DNSServerList        string            `json:",omitempty"`
+	DNSServerCompartment uint32            `json:",omitempty"`
+	ManagementIP         string            `json:",omitempty"`
+	AutomaticDNS         bool              `json:",omitempty"`
+}
+
+// SubnetInfo is assoicated with HNS network and represents a list
+// of subnets available to the network
+type SubnetInfo struct {
+	AddressPrefix  string            `json:",omitempty"`
+	GatewayAddress string            `json:",omitempty"`
+	Policies       []json.RawMessage `json:",omitempty"`
+}
+
+// MacPool is assoicated with HNS  network and represents a list
+// of macaddresses available to the network
+type MacPool struct {
+	StartMacAddress string `json:",omitempty"`
+	EndMacAddress   string `json:",omitempty"`
+}
+
+// DeleteHnsNetworkRequest describes request to delete the HNS network.
+type DeleteHnsNetworkRequest struct {
```

Comment:
Maybe we can combine the create and delete calls so that the user may call POST or DELETE and we just check the method type and process the request.

Reply:
We are not following that style in the other CNS APIs. I would prefer to keep it that way.